### DISCUSSION TITLE
Unable to set older kernel as default for BE; unsure why

### DISCUSSION THREAD

**Discussion by bkshoemaker** [zbm-dev/zfsbootmenu#659] [Questions]
I'm having trouble setting an older kernel as default for a boot environment, and haven't been able to figure out what's preventing it. The BE is a base dataset, not based on a snapshot or clone:

![image](https://github.com/user-attachments/assets/301a9818-7e8d-4f3c-8497-25972735216e)

I'm trying to set the 6.7 kernel as default here, but hitting ctrl-d here merely results in the header text changing from green to red. The BE still boots the newest 6.9 kernel by default.

![image](https://github.com/user-attachments/assets/07be0d9a-d556-4a34-853d-49a4be9e4d56)

![image](https://github.com/user-attachments/assets/7f4f1816-5798-4c18-b677-eaa0b6a9df5c)

This didn't generate any logs, and the only thing I've gleaned from the docs is that perhaps ZBM can't set the dataset read-write in order to add the kernel property. Does the red text denote an error here? Any ideas what's wrong/how to get this working?

**Comment by zdykstra** [comment]
The red text and r/w on the end indicate that it is able to import the pool read-write. The shortcut in the menus is just a helper to set org.zfsbootmenu:kernel on the dataset. If you want to set it directly for now, that should at least solve the immediate issue of pinning kernel 6.7. Refer to the ZFSBootMenu man page for the accepted values for that property.

**Comment by zdykstra** [comment]
I'll need to dig through the code again, but it's possible that the + sign in the kernel file name is confusing things.

**Comment by bkshoemaker** [comment]
Thanks for the quick response and clarifications.

I suspect you may be right about the plus sign breaking things -- I manually set the dataset property to "vmlinuz-6.7.12+bpo-amd64" and ZBM still defaulted to 6.9 on reboot, but then changing the property to just "6.7.12" made it boot 6.7 as expected.

**Reply by zdykstra** [reply]
Try escaping the + sign with a \ when using the full kernel name.
